crates/script/src/build.rs

```diff
@@ -188,8 +189,8 @@ impl PreprocessedState {
             MultiCompilerLanguage::FILE_EXTENSIONS,
         )
         .chain([target_path.to_path_buf()]);
-
-        let output = ProjectCompiler::new().files(sources_to_compile).compile(&project)?;
+        let revive: ReviveConfig = args.build.compiler.revive_opts.clone().into();
+        let output = ProjectCompiler::new().files(sources_to_compile).compile(&project, &revive)?;
```

Comment:
Did you check how it is implemented in Foundry-zkSync? It seems they have different approach. Please explain

Reply:
Yes i did, they follow a different path since they compilers repo doesnt use the default Project and when they compile their contracts they are using a custom ProjectCompileOutput where as for revive we use the default the revive config is there to show a spinner to alert developer when they are using revive"

Reply:
I am not sure if it is a good approach if we want to upstream our code. It suggests that if any other compiler needs to be added it will come here with its own config
so the function could look like `compile(project, config1, config2, config3 ...)`
WDYT?

Reply:
Not quite if another compiler is added we have to reconfigure the struct to be flexible such that it includes all compilers alternatively we can just pass the entire config struct which contains all info for all compilers